### ISSUE TITLE
Data Updates only after restarting Docker Container

### ISSUE THREAD

**Issue by Rocksteady0815** [tillsteinbach/WeConnect-mqtt#92]
Hi,
I noticed, that from time to time there are no updates received anymore.
I use the internal FHEM MQTT Broker.
I have to stop and restart the we-connect docker container to get new values.
I cannot see any log entries in the container.

Any idea, how to further investigate?

**Comment by tillsteinbach** [comment]
Hmm, is your expectation that you continuously receive data? Because you get only data if there are changes.
You can increase the log level by adding -v or -vv or even -vvv

**Comment by Rocksteady0815** [comment]
No, it is not my expectation to get continuously data.
Where can I increase the log level?

**Comment by tillsteinbach** [comment]
You can add `ADDITIONAL_PARAMETERS=-vv ` to the envs

**Comment by tillsteinbach** [comment]
just one minus, not two

**Comment by Rocksteady0815** [comment]
OK, that works.
Thanks.
I will check the logs in case it happens again.

**Comment by Rocksteady0815** [comment]
Hi,
I just noticed again, that data in FHEM are not updated.
The log seems to be OK:

![image](https://user-images.githubusercontent.com/10072618/155503329-e2b0ac4c-3152-49a3-ba2b-ff91c51c9dbe.png)

FHEM shows the last update two days ago, although I just used the car:

![image](https://user-images.githubusercontent.com/10072618/155503543-1ed74825-8f32-4f1f-b7f6-1286b9096717.png)

**Comment by tillsteinbach** [comment]
Can you somehow see in them if the broker is still connected? Also there is a topic`weconnect/0/mqtt/weconnectUpdated` what does that one say?

**Comment by Rocksteady0815** [comment]
I restarted the container again.
I will have a look on it, if it happens again.

Currently I can only see, that the weconnectUpdated seems to be on a different timezone. It is one hour back, than FHEM.

**Comment by tillsteinbach** [comment]
All times should be by default in UTC

**Comment by Rocksteady0815** [comment]
OK, again have no Update.

weconnectUpdated value shows  

2022-02-25T13:32:55+00:00
FHEM reading Time for this value is 14:32.

currentSOC_pct reading time in FHEM is 13:18

I leave the car 14:15

**Comment by tillsteinbach** [comment]
I assume you have MEZ timezone:
weconnectUpdated value 2022-02-25T13:32:55+00:00 (+00:00 shows that the time is UTC!) is today 14:32 in MEZ
this is matching
So what we see here is that you received data for the last time at 14:32 but the data from the car is from 13:18. Is that correctly summarized?
I currently cannot understand why that happens. If weconnectUpdated still changes after every interval it means that still data is being fetched. One thing I had in mind was that maybe FHEM is overloaded and looses messages, but if that would be the case it would work again if you start driving.
I have a different setup in my FHEM tests. I use mosquitto broker in between.

**Comment by Rocksteady0815** [comment]
Yes, correctly summarized.
I restarted the whole enviornment (docker, fhem, etc.).
Currently it worked again.

Only new values will be send by the corresponding topic, right?
That means, if new values are available, then only this topics get an update, not every topic..?

Maybe I should also try mosquitto.

**Comment by tillsteinbach** [comment]
no, only those topics get an update that changed. You can change that behaviour by adding `--republish-on-update` option. Then it publishes everything all the time. But this probably does not improve your situation if FHEM is really overloaded. One possibility would be then to reduce the number of topics. With `--topic-filter-regex` you could reduce the topics and filter out what you do not need.

**Comment by Rocksteady0815** [comment]
OK, thanks.
I really cannot imagine, that FHEM is overloaded.
I will keep an eye on it and maybe try to use mosquitto or experiment with RegEx.
Do I have to use the option within the docker-compose file?
Do you have an example?

**Comment by tillsteinbach** [comment]
If you use this compose file: https://github.com/tillsteinbach/WeConnect-mqtt/blob/main/docker/docker-compose.yml you can just add it to ADDITIONAL_PARAMETERS

**Comment by Rocksteady0815** [comment]
Currently, it seems to be another issue:

![image](https://user-images.githubusercontent.com/10072618/156513612-cef87581-efe1-4e60-9e72-33222f76b60f.png)

**Comment by tillsteinbach** [comment]
Can you copy the log for me? Also the content above your screenshot?

**Comment by Rocksteady0815** [comment]
Sure.
[putty.log](https://github.com/tillsteinbach/WeConnect-mqtt/files/8177913/putty.log)

**Comment by tillsteinbach** [comment]
It looks a bit like the container has no internet connection.

**Comment by tillsteinbach** [comment]
It should not, but maybe you hit an untested corner case. But still it looks like it cannot reach the servers

**Comment by Rocksteady0815** [comment]
Since I connected my RPI direct to LAN (not Wifi) almost every fault has been solved.
Everything is working stable for days...
